src/Microsoft.Framework.Configuration/ConfigurationKeyComparer.cs

```diff
@@ -8,14 +8,12 @@ namespace Microsoft.Framework.Configuration
 {
     public class ConfigurationKeyComparer : IComparer<string>
     {
-        private const char Separator = ':';
-
         public static ConfigurationKeyComparer Instance { get; } = new ConfigurationKeyComparer();
 
         public int Compare(string x, string y)
         {
-            var xParts = x?.Split(Separator) ?? new string[0];
-            var yParts = y?.Split(Separator) ?? new string[0];
+            var xParts = x?.Split(Constants.KeyDelimiter, StringSplitOptions.RemoveEmptyEntries) ?? new string[0];
```

Comment:
@davidfowl I already run all the ```ConfigurationKeyComparer``` tests and they passed, but there's one place i need change in ```ConfigurationBase``` https://github.com/aspnet/Configuration/blob/dev/src/Microsoft.Framework.Configuration/ConfigurationBase.cs#L72